test/sql/error/test_try_expression.test

```diff
@@ -0,0 +1,105 @@
+# name: test/sql/error/test_try_expression.test
+# group: [error]
+
+statement ok
+pragma enable_verification;
+
+# --- OPERATOR_CAST ---
+
+query I
+select TRY('abc'::INTEGER)
+----
+NULL
+
+# --- OPERATOR_NOT ---
+
+## FIXME: this should error?
+#query I
+#select ~error('test')
+#----
+#NULL
+
+query I
+select TRY(~error('test'))
+----
+NULL
+
+# --- OPERATOR_IS_NULL ---
+
+query I
+select TRY(error('test') IS NULL);
+----
+NULL
+
+
+# --- OPERATOR_IS_NOT_NULL ---
+
+query I
+select TRY(error('test') IS NOT NULL);
+----
+NULL
+
+# --- COMPARE_EQUAL ---
+
+query I
+select TRY(error('test') == 'abc')
+----
+NULL
+
+# --- COMPARE_EQUAL ---
+
+query I
+select TRY(error('test') == 'abc')
+----
+NULL
+
+# --- COMPARE_NOTEQUAL ---
+
+query I
+select TRY(error('test') != 'abc')
```

Comment:
Can we add a test where we `try` on a volatile function? e.g. `select try((random()*150)::TINYINT)`?

What about `try` with a subquery (e.g. `select try((select 'hello'::int))`)?

(both of these should probably error)

Reply:
Hmm good call, I understand that the volatility is problematic, because we can't repeat the same results.
I don't necessarily understand why the subquery one would be problematic? Is it because that potentially has other unwanted side effects?

Reply:
The expression in the subquery is not evaluated within the `ExpressionExecutor`, so the error will not be ignored

Reply:
Ah that makes sense, so we want to throw a BinderException when these are encountered?
I've got that working for volatile functions now

Reply:
Yes